include/deal.II/matrix_free/evaluation_template_factory.h

```diff
@@ -76,87 +66,34 @@ namespace internal
   struct FEFaceEvaluationFactory
   {
     static void
-    evaluate(const unsigned int n_components,
-             const MatrixFreeFunctions::ShapeInfo<VectorizedArrayType> &data,
-             const VectorizedArrayType *   values_array,
-             VectorizedArrayType *         values_quad,
-             VectorizedArrayType *         gradients_quad,
-             VectorizedArrayType *         hessians_quad,
-             VectorizedArrayType *         scratch_data,
-             const bool                    evaluate_values,
-             const bool                    evaluate_gradients,
-             const bool                    evaluate_hessians,
-             const unsigned int            face_no,
-             const unsigned int            subface_index,
-             const unsigned int            face_orientation,
-             const Table<2, unsigned int> &orientation_map);
+    evaluate(
+      const unsigned int                     n_components,
+      const EvaluationFlags::EvaluationFlags evaluation_flag,
+      const VectorizedArrayType *            values_dofs,
+      FEEvaluationBaseData<dim, Number, true, VectorizedArrayType> &eval);
 
     static void
-    integrate(const unsigned int n_components,
-              const MatrixFreeFunctions::ShapeInfo<VectorizedArrayType> &data,
-              VectorizedArrayType *         values_array,
-              VectorizedArrayType *         values_quad,
-              VectorizedArrayType *         gradients_quad,
-              VectorizedArrayType *         hessians_quad,
-              VectorizedArrayType *         scratch_data,
-              const bool                    integrate_values,
-              const bool                    integrate_gradients,
-              const bool                    integrate_hessians,
-              const unsigned int            face_no,
-              const unsigned int            subface_index,
-              const unsigned int            face_orientation,
-              const Table<2, unsigned int> &orientation_map);
+    integrate(
+      const unsigned int                     n_components,
+      const EvaluationFlags::EvaluationFlags integration_flag,
+      VectorizedArrayType *                  values_dofs,
+      FEEvaluationBaseData<dim, Number, true, VectorizedArrayType> &eval);
 
-    static bool
+    static void
     gather_evaluate(
       const unsigned int                          n_components,
-      const std::size_t                           n_face_orientations,
```

Comment:
Why could you remove this?

Reply:
They just moved down to their own class - I thought it is somewhat easier to understand if we have an `evaluate` and `integrate` for each of the cells, faces, and face + vector access cases. See the class `FEFaceEvaluationGatherFactory` below. (This way I can also reduce the number of template arguments to where I really need `Number` and `VectorizedArrayType`.)